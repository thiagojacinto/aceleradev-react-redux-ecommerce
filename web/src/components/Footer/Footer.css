@import url("../../styles/gradientAnimation.css");

footer.footer {
  height: 10vh;
}

footer .container {
  height: inherit;

  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-evenly;
}

.footer__description {
  font-size: 1em;
}

.footer__description a {
  background-color: #835a62;
  color: #ffcdb2;
  box-shadow: 0 5px 0 #6d6875;
  padding: 0.5em 1em;
  margin: 0 0.6em;
  text-decoration: none;
  text-transform: uppercase;
  border-radius: 0.5em;
}

.footer__description a:hover {
  background-color: #6d4a51;
  cursor: pointer;
}

.footer__description a:active {
  box-shadow: none;
  top: 0.3em;
}
