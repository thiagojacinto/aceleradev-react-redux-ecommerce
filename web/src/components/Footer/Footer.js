import React from "react";
import { TwitterSquare, LinkedinIn, GithubSquare } from "@styled-icons/fa-brands";

export const Footer = ({
  authorName,
  githubLink,
  twitterLink,
  linkedInLink,
}) => {
  return (
    <footer className="footer">
      <div className="container">
        <section className="footer__description">
          <span>
            Criado por
            <a href={githubLink} target="_blank" rel="noopener noreferrer">
              {" "}
              {authorName}
            </a>
          </span>
        </section>

        <section className="footer__social-links">
        <a
              className="footer__social__github"
              href={githubLink}
              target="_blank"
              rel="noopener noreferrer"
            >
              <GithubSquare size="30" />
            </a>

          {twitterLink && (
            <a
              className="footer__social__twitter"
              href={twitterLink}
              target="_blank"
              rel="noopener noreferrer"
            >
              <TwitterSquare size="30" />
            </a>
          )}
          {linkedInLink !== undefined && (
            <a
              className="footer__social__linkedin"
              href={linkedInLink}
              target="_blank"
              rel="noopener noreferrer"
            >
              <LinkedinIn size="30" />
            </a>
          )}
        </section>
      </div>
    </footer>
  );
};

export default Footer;