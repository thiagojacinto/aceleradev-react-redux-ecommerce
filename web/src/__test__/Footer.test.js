import React from "react";
import { render, cleanup, getByTestId } from "@testing-library/react";

import Footer from "../components/Footer";

afterEach(cleanup);

const footerMock = {
  author: "author",
  github: "github",
  twitter: "twitter",
  linkedIn: "linkedIn",
};

describe("Testing Footer component ...", () => {
  test("Should take a snapshot", () => {
    const { asFragment } = render(
      <Footer
        authorName={footerMock.author}
        githubLink={footerMock.github}
        twitterLink={footerMock.twitter}
        linkedInLink={footerMock.linkedIn}
      />
    );

    expect(asFragment(<Footer />)).toMatchSnapshot();
  });

  test("Render author sign in Footer", () => {
    const author = footerMock.author;
    const { getByText } = render(<Footer authorName={author} />);
    const authorSign = getByText(/author/i);
    expect(authorSign).toBeInTheDocument();
  });
});
