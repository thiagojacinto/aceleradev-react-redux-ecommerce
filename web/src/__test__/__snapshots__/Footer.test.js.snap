// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer component ... Should take a snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <section
        class="footer__description"
      >
        <span>
          Criado por
          <a
            href="github"
            rel="noopener noreferrer"
            target="_blank"
          >
             author
          </a>
        </span>
      </section>
      <section
        class="footer__social-links"
      >
        <a
          class="footer__social__github"
          href="github"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="sc-AxjAm dVOzqR"
            fill="currentColor"
            focusable="false"
            height="30"
            viewBox="0 0 448 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM277.3 415.7c-8.4 1.5-11.5-3.7-11.5-8 0-5.4.2-33 .2-55.3 0-15.6-5.2-25.5-11.3-30.7 37-4.1 76-9.2 76-73.1 0-18.2-6.5-27.3-17.1-39 1.7-4.3 7.4-22-1.7-45-13.9-4.3-45.7 17.9-45.7 17.9-13.2-3.7-27.5-5.6-41.6-5.6-14.1 0-28.4 1.9-41.6 5.6 0 0-31.8-22.2-45.7-17.9-9.1 22.9-3.5 40.6-1.7 45-10.6 11.7-15.6 20.8-15.6 39 0 63.6 37.3 69 74.3 73.1-4.8 4.3-9.1 11.7-10.6 22.3-9.5 4.3-33.8 11.7-48.3-13.9-9.1-15.8-25.5-17.1-25.5-17.1-16.2-.2-1.1 10.2-1.1 10.2 10.8 5 18.4 24.2 18.4 24.2 9.7 29.7 56.1 19.7 56.1 19.7 0 13.9.2 36.5.2 40.6 0 4.3-3 9.5-11.5 8-66-22.1-112.2-84.9-112.2-158.3 0-91.8 70.2-161.5 162-161.5S388 165.6 388 257.4c.1 73.4-44.7 136.3-110.7 158.3zm-98.1-61.1c-1.9.4-3.7-.4-3.9-1.7-.2-1.5 1.1-2.8 3-3.2 1.9-.2 3.7.6 3.9 1.9.3 1.3-1 2.6-3 3zm-9.5-.9c0 1.3-1.5 2.4-3.5 2.4-2.2.2-3.7-.9-3.7-2.4 0-1.3 1.5-2.4 3.5-2.4 1.9-.2 3.7.9 3.7 2.4zm-13.7-1.1c-.4 1.3-2.4 1.9-4.1 1.3-1.9-.4-3.2-1.9-2.8-3.2.4-1.3 2.4-1.9 4.1-1.5 2 .6 3.3 2.1 2.8 3.4zm-12.3-5.4c-.9 1.1-2.8.9-4.3-.6-1.5-1.3-1.9-3.2-.9-4.1.9-1.1 2.8-.9 4.3.6 1.3 1.3 1.8 3.3.9 4.1zm-9.1-9.1c-.9.6-2.6 0-3.7-1.5s-1.1-3.2 0-3.9c1.1-.9 2.8-.2 3.7 1.3 1.1 1.5 1.1 3.3 0 4.1zm-6.5-9.7c-.9.9-2.4.4-3.5-.6-1.1-1.3-1.3-2.8-.4-3.5.9-.9 2.4-.4 3.5.6 1.1 1.3 1.3 2.8.4 3.5zm-6.7-7.4c-.4.9-1.7 1.1-2.8.4-1.3-.6-1.9-1.7-1.5-2.6.4-.6 1.5-.9 2.8-.4 1.3.7 1.9 1.8 1.5 2.6z"
              fill="currentColor"
            />
          </svg>
        </a>
        <a
          class="footer__social__twitter"
          href="twitter"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="sc-AxjAm dVOzqR"
            fill="currentColor"
            focusable="false"
            height="30"
            viewBox="0 0 448 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-48.9 158.8c.2 2.8.2 5.7.2 8.5 0 86.7-66 186.6-186.6 186.6-37.2 0-71.7-10.8-100.7-29.4 5.3.6 10.4.8 15.8.8 30.7 0 58.9-10.4 81.4-28-28.8-.6-53-19.5-61.3-45.5 10.1 1.5 19.2 1.5 29.6-1.2-30-6.1-52.5-32.5-52.5-64.4v-.8c8.7 4.9 18.9 7.9 29.6 8.3a65.447 65.447 0 01-29.2-54.6c0-12.2 3.2-23.4 8.9-33.1 32.3 39.8 80.8 65.8 135.2 68.6-9.3-44.5 24-80.6 64-80.6 18.9 0 35.9 7.9 47.9 20.7 14.8-2.8 29-8.3 41.6-15.8-4.9 15.2-15.2 28-28.8 36.1 13.2-1.4 26-5.1 37.8-10.2-8.9 13.1-20.1 24.7-32.9 34z"
              fill="currentColor"
            />
          </svg>
        </a>
        <a
          class="footer__social__linkedin"
          href="linkedIn"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="sc-AxjAm dVOzqR"
            fill="currentColor"
            focusable="false"
            height="30"
            viewBox="0 0 448 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 01107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
              fill="currentColor"
            />
          </svg>
        </a>
      </section>
    </div>
  </footer>
</DocumentFragment>
`;
